presto-raptor/src/test/java/com/facebook/presto/raptor/storage/TestOrcFileRewriter.java

```diff
@@ -413,9 +428,225 @@ public void testUncompressedSize(boolean useOptimizedOrcWriter)
         }
 
         File newFile = new File(temporary, randomUUID().toString());
-        OrcFileInfo info = new OrcRecordFileRewriter().rewrite(file, newFile, new BitSet());
+        OrcFileInfo info = createFileRewriter(useOptimizedOrcWriter).rewrite(getColumnTypes(columnIds, columnTypes), file, newFile, new BitSet());
         assertEquals(info.getRowCount(), 3);
-        assertEquals(info.getUncompressedSize(), 55);
+        assertBetweenInclusive(info.getUncompressedSize(), 55L, 55L * 2);
+    }
+
+    /**
+     * The following test add or drop different columns
+     */
+    @Test(dataProvider = "useOptimizedOrcWriter")
+    public void testRewriterDropThenAddDifferentColumns(boolean useOptimizedOrcWriter)
+            throws Exception
+    {
+        TypeRegistry typeRegistry = new TypeRegistry();
+        DBI dbi = new DBI("jdbc:h2:mem:test" + System.nanoTime());
+        dbi.registerMapper(new TableColumn.Mapper(typeRegistry));
+        Handle dummyHandle = dbi.open();
+        File dataDir = Files.createTempDir();
+
+        StorageManager storageManager = createOrcStorageManager(dbi, dataDir, useOptimizedOrcWriter);
+
+        List<Long> columnIds = ImmutableList.of(3L, 7L);
+        List<Type> columnTypes = ImmutableList.of(BIGINT, createVarcharType(20));
+
+        File file = new File(temporary, randomUUID().toString());
+        try (FileWriter writer = createFileWriter(columnIds, columnTypes, file, false, useOptimizedOrcWriter)) {
+            List<Page> pages = rowPagesBuilder(columnTypes)
+                    .row(2L, "2")
+                    .build();
+            writer.appendPages(pages);
+        }
+
+        // Add a column
+        File newFile1 = new File(temporary, randomUUID().toString());
+        OrcFileInfo info = createFileRewriter(useOptimizedOrcWriter).rewrite(
+                getColumnTypes(ImmutableList.of(3L, 7L, 10L), ImmutableList.of(BIGINT, createVarcharType(20), DOUBLE)),
+                file,
+                newFile1,
+                new BitSet(5));
+        assertEquals(info.getRowCount(), 1);
```

Comment:
Assert that the file does not change. But I suggest we change the test to let it actually delete rows. The reason is as followed:

Note that in practice, arguments pass to rewriter is guaranteed to have non-zero delete set (`OrcStorageManager::createShardRewriter()`). It's totally valid to assert in `FileRewriter` that the delete set is not empty. And the assert will break your test here since you are not deleting anything.